marshal/bin/agent.py

```diff
@@ -263,7 +263,9 @@ def main():
             rootfs_s3_etag = None
 
             for artifact in manifest["artifacts"]:
-                if artifact["type"] == "application/vnd.cray.image.rootfs.squashfs":
+                if artifact["type"] == "application/vnd.cray.image.rootfs.squashfs" and \
+                    'link' in artifact and artifact['link'] is not None and \
+                    'path' in artifact['link'] is not None and 'etag' in artifact['link'] is not None:
```

Comment:
As long as you know both of these keys exist this should be fine because the keys will exist so the key lookups will work
```suggestion
                    'path' in artifact['link'] and 'etag' in artifact['link']:
```

Reply:
@aasha-hpe fixed this with the suggestion